src/pipeline_v3/core/fingerprint.py

```
@@ -69,82 +71,87 @@
-        
+
         if not source_path.exists():
             raise FileNotFoundError(f"Source file not found: {source}")
-        
+
         # Read file content
         try:
             content = source_path.read_bytes()
             content_hash = hashlib.sha256(content).hexdigest()
         except Exception as e:
             raise Exception(f"Failed to read file {source}: {e}")
```

Pylint (3.3.7)

[warning] Consider explicitly re-raising using 'raise Exception(f'Failed to read file {source}: {e}') from e'

(W0707)